### PR TITLE
Contact Page and Email Subscribe

### LINKED ISSUE #11: Footer: email sign up
Hook up the email sign up in the footer to salesforce. This may not be necessary for the Feb 26 demo though (?) so putting as low priority

From Lou: look up examples in the cpal-site repo or seda-site repo

### LINKED ISSUE #31: Contact page
Requirements:
1) Form that includes the following fields:
- First name
- Last name
- Email 
- Subject
- Message
- Checkbox to sign up for email updates - checking this box adds the user's email address to the email list (same as the email sign up form in the footer)

2) Form submissions should be sent to this email address - analytics@childpovertyactionlab.org

3) See this contact form as an example (except we don't need the "I am reaching out" dropdown at the top):
image

4) Info about Netlify forms: https://docs.netlify.com/forms/setup/

### PR DESCRIPTION
**Pull request by touchRED**
Closes #11 and closes #31.

Notes: 

- The main addition here is the `subscribe` netlify function and its corresponding functionality in the footer and on the contact page
- The netlify function accepts an email and posts it to the list of leads in Salesforce. This might need a bit of work, but I wasn't sure how to proceed with QA locally since the function needs to be recognized by netlify. 
- The footer form is setup to use formik/yup as requested, and contains success/error transitions for feedback.

### CODE REVIEW

File: netlify/functions/subscribe.js [hunk 1]
```diff
@@ -0,0 +1,36 @@
+const https = require('https')
+
+exports.handler = async function(event, context) {
+    const { email } = JSON.parse(event.body)
+
+    const data = {
+        email: email,
+        oid: "00D1U00000110AJ"
```
Comment (amygroshek) [inline netlify/functions/subscribe.js]: I *think* we probably want to store this in the `.env` file and pass it to the netlify function when we call it from the gatsby UI. It's a marginal security concern and I know the default SF froms expose this ID but we have more control and so may as well handle it better. You'll want to add this value to the environment values in the Netlify UI as well.

**Review by amygroshek — CHANGES_REQUESTED** [review]
@touchRED This looks great. 👍🏻  

* Please also uncomment and update the line in `netlify.toml` that tells Netlify there is a functions dir to build. 
* Client's `oid` in the netlify function will be exposed in the github repo, remove and access from `.env` file instead.
* I think it also makes sense to spend the time setting up [netlify dev](https://www.netlify.com/products/dev/) or [netlify-lambda](https://github.com/netlify/netlify-lambda) for local testing of the form at the same time. Mostly to cut down on having to deploy a bunch of times to test. 
* It looks like I didn't do this when I implemented this contact form previously, but we should have a honeypot in both forms and verify that it is not filled in before carrying out any submit actions. The honeypot field is in the contact form but (correct me if I'm wrong) I don't see it being handled by the form validation or the submit action. And I don't see one in the signup in the footer. This is important on Netlify because a bot could hit the site and push the client over their free limit. If we were using the default submit action for form handling, Netlify would detect and process the honeypot field, but because we're using JavaScript we need to check it ourselves. 

A few other UX comments: 
* I would love it if I could get a valid entry checkmark for the email input as well as the others. 
Screen Shot 2021-03-15 at 5 50 19 PM

* I would love a spinner during form submission, maybe just right beside the submit button or inside of it, while the netlify function and then the contact form submit are processing. Have a look at this page: https://edopportunity.org/get-the-data/
* It seems like I get a success message even when the submit is clearly failing. 
Screen Shot 2021-03-15 at 5 48 49 PM

Other nit-picky stuff: 
* I'm wondering if we can disable the default validation for the footer signup and do validation like we do in the contact form instead? 
Screen Shot 2021-03-15 at 5 58 31 PM
* I like the feedback on the footer submit form, but I find it a bit disorienting that the input completely disappears after I successfully submit. Can we have it persist and just change the styling a bit to suggest "disabled"? Maybe use a lighter gray for the underline and button? 
Screen Shot 2021-03-15 at 5 59 21 PM



@laurendoone If Sasha has the cycles maybe we can have him look at styling the form inputs? If not, I think we should probably emulate the styling of the subscribe input for the simple text inputs (not the textarea).

**Comment by touchRED** [comment]
@lougroshek added some more updates here – using `netlify dev` now as a develop command and everything seems to be working swimmingly – lmk where this should sit in our `package.json` commands, I'll be fixing up your other requested changes in the meantime! feel free to drop anything else here

**Comment by touchRED** [comment]
@lougroshek some more updates addressing the spinner, submission feedback/opacity, and validation behavior in the footer

**Comment by amygroshek** [comment]
@touchRED Much improved! Thank you!

**Pull request merged**